Emby.Server.Implementations/Data/BaseSqliteRepository.cs

```diff
@@ -1,183 +1,76 @@
-                    };
-
-                    if (CacheSize.HasValue)
-                    {
-                        queries.Add("PRAGMA cache_size=" + CacheSize.Value.ToString(CultureInfo.InvariantCulture));
-                    }
-
-                    if (EnableTempStoreMemory)
-                    {
-                        queries.Add("PRAGMA temp_store = memory");
-                    }
-                    else
-                    {
-                        queries.Add("PRAGMA temp_store = file");
-                    }
-
-                    foreach (var query in queries)
-                    {
-                        db.Execute(query);
-                    }
-                }
-                catch
-                {
-                    using (db)
-                    {
-
-                    }
-
-                    throw;
-                }
+            if (string.IsNullOrWhiteSpace(_defaultWal))
+            {
+                _defaultWal = WriteConnection.Query("PRAGMA journal_mode").SelectScalarString().First();
 
-                _connection = new ManagedConnection(db, false);
+                Logger.LogInformation("Default journal_mode for {0} is {1}", DbFilePath, _defaultWal);
+            }
 
-                return _connection;
+            if (EnableTempStoreMemory)
+            {
+                WriteConnection.Execute("PRAGMA temp_store = memory");
+            }
+            else
+            {
+                WriteConnection.Execute("PRAGMA temp_store = file");
             }
-        }
 
-        public IStatement PrepareStatement(ManagedConnection connection, string sql)
-        {
-            return connection.PrepareStatement(sql);
+            return new ManagedConnection(WriteConnection, WriteLock);
         }
 
-        public IStatement PrepareStatementSafe(ManagedConnection connection, string sql)
-        {
-            return connection.PrepareStatement(sql);
-        }
+        public IStatement PrepareStatement(ManagedConnection connection, string sql)
+            => connection.PrepareStatement(sql);
 
         public IStatement PrepareStatement(IDatabaseConnection connection, string sql)
-        {
-            return connection.PrepareStatement(sql);
-        }
+            => connection.PrepareStatement(sql);
 
-        public IStatement PrepareStatementSafe(IDatabaseConnection connection, string sql)
-        {
-            return connection.PrepareStatement(sql);
-        }
-
-        public List<IStatement> PrepareAll(IDatabaseConnection connection, IEnumerable<string> sql)
-        {
-            return PrepareAllSafe(connection, sql);
-        }
-
-        public List<IStatement> PrepareAllSafe(IDatabaseConnection connection, IEnumerable<string> sql)
-        {
-            return sql.Select(connection.PrepareStatement).ToList();
-        }
+        public IEnumerable<IStatement> PrepareAllSafe(IDatabaseConnection connection, IEnumerable<string> sql)
```

Comment:
why are you keeping `PrepareAllSafe` instead of `PrepareAll`? other functions with `Safe` in them got dropped...

Reply:
Fixed